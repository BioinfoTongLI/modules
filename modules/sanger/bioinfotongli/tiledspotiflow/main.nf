container_version = "0.5.1"

params.debug = false

process BIOINFOTONGLI_TILEDSPOTIFLOW {
    debug params.debug
    tag "${meta.id}"

    label "gpu"
    label "process_medium"

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        "quay.io/bioinfotongli/tiled_spotiflow:${container_version}":
        "quay.io/bioinfotongli/tiled_spotiflow:${container_version}"}"
    containerOptions = {
        workflow.containerEngine == "singularity" ? "--cleanenv --nv":
        ( workflow.containerEngine == "docker" ? "--gpus all": null )
    }
    publishDir params.out_dir + "/spotiflow_peaks"

    input:
    tuple val(meta), val(x_min), val(y_min), val(x_max), val(y_max), path(image), val(ch_ind)
    
    output:
    tuple val(meta), val(ch_ind), path("${out_dir}/${out_name}"), emit: peaks
    path "versions.yml"           , emit: versions

    script:
    def args = task.ext.args ?: ''
    out_dir = "${meta.id}_ch_${ch_ind}"
    out_name = "ch_${ch_ind}_peaks_Y${y_min}_X${x_min}.csv"
    """
    /opt/conda/bin/python /scripts/Spotiflow_call_peaks.py run \
        -image_path ${image} \
        -out_dir "${out_dir}" \
        -x_min ${x_min} \
        -y_min ${y_min} \
        -x_max ${x_max} \
        -y_max ${y_max} \
        -ch_ind ${ch_ind} \
        -out_name "${out_name}" \
        ${args}
    
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        : \$(echo \$(/opt/conda/bin/python /scripts/Spotiflow_call_peaks.py version 2>&1) | sed 's/^.*Spotiflow_call_peaks.py //; s/Using.*\$//' ))
    END_VERSIONS
    """
}