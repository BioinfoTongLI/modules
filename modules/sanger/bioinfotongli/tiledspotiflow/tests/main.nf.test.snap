{
    "mouse_heart - tiff - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        "mouse_heart_ch_0_peaks_Y0_X0.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,66e09e2d1d4c11e7897cf031180a176e"
                ],
                "peaks": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        "mouse_heart_ch_0_peaks_Y0_X0.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,66e09e2d1d4c11e7897cf031180a176e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-02-07T10:50:39.197147856"
    },
    "mouse_heart - tiff": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "mouse_heart",
                            "ch_ind": 0
                        },
                        "mouse_heart_ch_0_peaks_Y0_X0.csv:md5,a1a930de917429f510577aa17970aaa6"
                    ]
                ],
                "1": [
                    "versions.yml:md5,66e09e2d1d4c11e7897cf031180a176e"
                ],
                "peaks": [
                    [
                        {
                            "id": "mouse_heart",
                            "ch_ind": 0
                        },
                        "mouse_heart_ch_0_peaks_Y0_X0.csv:md5,a1a930de917429f510577aa17970aaa6"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,66e09e2d1d4c11e7897cf031180a176e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-02-07T10:50:04.648448036"
    },
    "mouse_heart - tiff - 500": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "mouse_heart",
                            "ch_ind": 0
                        },
                        "mouse_heart_ch_0_peaks_Y0_X0.csv:md5,b751ba2081cc127d53c9457f5b15749d"
                    ]
                ],
                "1": [
                    "versions.yml:md5,66e09e2d1d4c11e7897cf031180a176e"
                ],
                "peaks": [
                    [
                        {
                            "id": "mouse_heart",
                            "ch_ind": 0
                        },
                        "mouse_heart_ch_0_peaks_Y0_X0.csv:md5,b751ba2081cc127d53c9457f5b15749d"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,66e09e2d1d4c11e7897cf031180a176e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-02-07T10:50:26.78446238"
    }
}